src/main/java/org/agileware/test/PropertiesTester.java

```diff
@@ -226,47 +231,127 @@ private void testField(Object instance, Class<?> type, Field field, Object value
-            	Assert.assertEquals("Null field test failed on `" + field.getName() + "`", null, getter.invoke(instance));
-            }
+            testSetterGetterInteraction(instance, field, setter, getter, value);
         } else {
             if (setter != null) {
-            	if (value == null)
-            		value = this.getInstance(field.getType());
-                field.setAccessible(true);
-                setter.invoke(instance, value);
-                Assert.assertEquals("Field test failed on `" + field.getName() + "`", value, field.get(instance));
-                if (!field.getType().isPrimitive()) {
-                	setter.invoke(instance, field.getType().cast(null));
-                	Assert.assertEquals("Null field test failed on `" + field.getName() + "`", null, field.get(instance));
-                }
+                testSetter(instance, field, setter, value);
             }
             if (getter != null) {
-            	if (value == null)
-            		value = this.getInstance(field.getType());
-                field.setAccessible(true);
-                field.set(instance, value);
-                Assert.assertEquals("Field test failed on `" + field.getName() + "`", value, getter.invoke(instance));
-                if (!field.getType().isPrimitive()) {
-                	field.set(instance, null);
-                	Assert.assertEquals("Null field test failed on `" + field.getName() + "`", null, getter.invoke(instance));
-                }
+                testGetter(instance, field, getter, value);
+            }
+        }
+    }
+
+    /**
+     * This method tests the interaction between a fields setter and getter methods
+     * @param instance the instance on which to invoke calls
+     * @param field the field under test
+     * @param setter the setter method to invoke
+     * @param getter the getter method to invoke
+     * @param value the value passed to setter method, and expected to return from getter method invocation
+     * @throws InstantiationException
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
+     * @throws NoSuchMethodException
+     * @throws InvocationTargetException 
+     */
+    private void testSetterGetterInteraction(Object instance, Field field, Method setter, Method getter, Object value) throws InstantiationException, IllegalAccessException, IllegalArgumentException, NoSuchMethodException, InvocationTargetException {
+        Object getterInvocationResult;
+        if (value == null){
+            value = this.getInstance(field.getType());
+        }
+        setter.invoke(instance, value);
+        getterInvocationResult = getter.invoke(instance);
+        if(Optional.class.isInstance(getterInvocationResult) && !Optional.class.isInstance(value)){
+            Assert.assertTrue("Setter-Getter interaction for field test failed on `" + field.getName() + "`, expected none empty Optional", Optional.class.cast(getterInvocationResult).isPresent());

```

Comment:
Probably these messages should be better described: "Getter-Setter interaction..." doesn't sound right to me... What about "Accessor methods test failed on field xxx ..."